py/picca/io.py

```diff
@@ -1375,116 +1088,118 @@ def read_from_minisv_desi(in_dir,
+                spec_data[color] = spec
             except OSError:
-                userprint(f"error when reading {spectrograph}-band data")
+                userprint(f"ERROR: when reading {color}-band data")
 
         hdul.close()
         plate_spec = int(f"{tile_spec}{petal_spec}")
 
-        select=(plate==plate_spec)&(night==night_spec)
-        print('\nThis is tile {}, petal {}, night {}'.format(tile_spec,petal_spec,night_spec))
-        thingid_qsos = thingid[select]
-        plate_qsos = plate[select]
-        night_qsos = night[select]
-        fiberid_qsos = fiberid[select]
-        for t, p, m, f in zip(thingid_qsos,plate_qsos,night_qsos,fiberid_qsos):
-            w_t = in_thingids == t
-            if w_t.sum() == 0:
-                print(f"\nError reading thingid {t}\n")
-                print(f"spectra : {spec}\n")
-                print(f"plate_spec : {plate_spec}\n")
-                continue
+        select = ( (catalog['TILEID'] == tile_spec) &
+                   (catalog['PETAL_LOC'] == petal_spec) & 
+                   (catalog['NIGHT']  == night_spec) )
+        userprint(f'This is tile {tile_spec}, petal {petal_spec}, night {night_spec}')
+
+        #-- Loop over quasars in catalog inside this tile-petal
+        for entry in catalog[select]:
 
-            forest=None
+            #-- Find which row in tile contains this quasar
+            w_t = np.where(targetid_spec == entry['TARGETID'])[0][0]
+
+            #-- Loop over three spectrograph arms and coadd fluxes
+            forest = None
             for spec in spec_data.values():
                 ivar = spec['IV'][w_t]
-                flux = (ivar*spec['FL'][w_t]).sum(axis=0)
-                ivar = ivar.sum(axis=0)
-                w = ivar > 0.
-                flux[w] /= ivar[w]
+                flux = spec['FL'][w_t]
 
                 if pk1d is not None:
-                    reso_sum = spec['RESO'][w_t].sum(axis=0)
+                    reso_sum = spec['RESO'][w_t]
                     reso_in_km_per_s = np.real(spectral_resolution_desi(
-                        reso_sum,spec['log_lambda']))
+                        reso_sum, spec['log_lambda']))
                     exposures_diff = np.zeros(spec['log_lambda'].shape)
                 else:
                     reso_in_km_per_s = None
                     exposures_diff = None
 
-                forest_temp = Forest(spec['log_lambda'],flux,ivar,t,ra[w_t][0],
-                    dec[w_t][0],z_table[t], p, m, f, order,
+                forest_temp = Forest(spec['log_lambda'], flux, ivar,
+                    entry['TARGETID'],
+                    entry['RA'], entry['DEC'], entry['Z'],
+                    entry['TILEID'], entry['NIGHT'], entry['FIBER'],  
                     exposures_diff, reso_in_km_per_s)
                 if forest is None:
-                    forest = copy.deepcopy(forest_temp)
+                    forest = forest_temp
```

Comment:
why are you not using copy here but using it on line 1057?

Reply:
We should probably use copy just to make sure. I'm adding it back